### DISCUSSION TITLE
Modify a lengthy, auto-generated component name?

### DISCUSSION THREAD

**Discussion by gantsevdenis** [fastapi/fastapi#12081] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import APIRouter, UploadFile, File
router = APIRouter(prefix="/abc/{arg1}")
...
@router.post("/upload", response_class=ORJSONResponse, tags=["mytag"], openapi_extra=dict(operationId="uploadcsv")
def handler_upload_csv(arg, file: UploadFile = File())
    pass
```

### Description

This generates a very long name for the component, like: "Body_handler_upload_csv_arg1__post"; Given the fact that I auto-generate openapi client for my API, I would like to change that name to make it more digestible? I tried "File(title=...)", but didn't work, and I can't find the corresponding documentation? I am actually not even sure how to word the question in order to google it

### Operating System

Linux

### FastAPI Version

0.111.1

### Pydantic Version

2.8.2

### Python Version

3.9.13

### Additional Context

using https://github.com/openapi-generators/openapi-python-client to generate my client

**Comment by YuriiMotov** [comment, marked as answer]
The only way I see is to correct openapi schema replacing long names to preferred names:

```
import json

from fastapi import FastAPI
from fastapi.openapi.utils import get_openapi


app = FastAPI()


from fastapi import APIRouter, UploadFile, File

router = APIRouter(prefix="/abc/{arg1}")


@router.post("/upload")
def handler_upload_csv(arg, file: UploadFile = File()):
    pass
    

app.include_router(router=router)



def custom_openapi():
    if app.openapi_schema:
        return app.openapi_schema
    openapi_schema = get_openapi(
        title=app.title,
        version=app.version,
        summary=app.summary,
        description=app.description,
        routes=app.routes,
    )
    openapi_json = json.dumps(openapi_schema)
    openapi_json = openapi_json.replace('Body_handler_upload_csv_abc__arg1__upload_post', 'MyFile')

    app.openapi_schema = json.loads(openapi_json)
    return app.openapi_schema

app.openapi = custom_openapi
```

**Reply by gantsevdenis** [reply]
that's ultimately what I did, thanks for the help!
